crates/bitwarden-core/src/key_management/crypto.rs

```diff
@@ -557,6 +570,51 @@ pub(super) fn verify_asymmetric_keys(
     })
 }
 
+/// A new signing key pair along with the signed public key
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct MakeUserSigningKeysResponse {
+    /// Base64 encoded verifying key
+    verifying_key: String,
+    /// Signing key, encrypted with a symmetric key (user key, org key)
+    signing_key: EncString,
+    /// The user's public key, signed by the signing key
+    signed_public_key: SignedPublicKey,
+}
+
+/// Makes a new set of signing keys for a user, which should only be done during
+/// once. This also signs the public key with the signing key
+/// and returns the signed public key.
+pub fn make_user_signing_keys_for_enrollment(
+    client: &Client,
+) -> Result<MakeUserSigningKeysResponse, CryptoError> {
+    let key_store = client.internal.get_key_store();
+    let ctx = key_store.context();
+
+    #[allow(deprecated)]
+    let wrapping_key = ctx
+        .dangerous_get_symmetric_key(SymmetricKeyId::User)
+        .map_err(|_| CryptoError::InvalidKey)?;
+
+    // Make new keypair and sign the public key with it
+    let signature_keypair =
+        SigningKey::make(SignatureAlgorithm::Ed25519).map_err(|_| CryptoError::InvalidKey)?;
+    let signed_public_key = ctx.make_signed_public_key(
+        AsymmetricKeyId::UserPrivateKey,
+        SigningKeyId::UserSigningKey,
+    )?;
+
+    Ok(MakeUserSigningKeysResponse {
+        verifying_key: STANDARD.encode(signature_keypair.to_verifying_key().to_cose()),
+        // This needs to be changed to use the correct COSE content format before rolling out to
+        // users: https://bitwarden.atlassian.net/browse/PM-22189
+        signing_key: signature_keypair.to_cose().encrypt_with_key(wrapping_key)?,
```

Comment:
We can remove the direct access to the user key and the `#[allow(deprecated)]` by using the new api here
```suggestion
        signing_key: signature_keypair
            .to_cose()
            .encrypt(&mut ctx, SymmetricKeyId::User)?,
```

Reply:
Fixed.